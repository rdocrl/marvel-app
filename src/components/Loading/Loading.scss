.loading {
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  background-color: gray;
  display: flex;

  &__logo {
    width: 200px;
    animation: loading 2s infinite;
    margin: 0 auto;
  }
}

@keyframes loading {
  0% {
    width: 100px;
  }
  50% {
    width: 200px;
  }

  100% {
    width: 100px;
  }
}
