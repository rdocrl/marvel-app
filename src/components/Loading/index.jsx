import logo from '../../assets/logo.svg';
import './Loading.scss';

const Loading = () => {
  return (
    <div className="loading">
      <img className="loading__logo" src={logo} alt="Loading..." />
    </div>
  );
};

export default Loading;
