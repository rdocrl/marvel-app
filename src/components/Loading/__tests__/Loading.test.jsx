import { render } from '@testing-library/react';
import Loading from '../';

describe('<Loading /> tests', () => {
  describe('Snapshots', () => {
    it('Should render', () => {
      const { asFragment } = render(<Loading />);
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
