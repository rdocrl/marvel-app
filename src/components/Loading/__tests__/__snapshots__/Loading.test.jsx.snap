// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading /> tests Snapshots Should render 1`] = `
<DocumentFragment>
  <div
    class="loading"
  >
    <img
      alt="Loading..."
      class="loading__logo"
      src="logo.svg"
    />
  </div>
</DocumentFragment>
`;
